base_jsonify/README.rst

```diff
@@ -79,6 +79,95 @@ you can pass a callable or the name of a method on the model:
 Also the module provide a method "get_json_parser" on the ir.exports object
 that generate a parser from an ir.exports configuration.
 
+Further features are available for advanced uses.
+It defines a simple "resolver" model that has a "python_code" field and an eval
+function so that arbitrary functions can be configured to transform fields,
+or process the resulting dictionary.
+It is also to specify a lang to extract the translation of any given field.
```

Comment:
I don't get why you need this generate translations: you should call `jsonify` with the proper lang in the ctx :thinking:

Reply:
@simahawk As I understand it, to have different lang at same time into the result of jsonify

Reply:
We are working with an API that wants all translations at the same time; so if you want to pass the description_sale, you need also to pass description_sale_fr_BE, description_sale_nl_BE, ...

In a way that could be avoided by externalising the lang, so instead you would have one export per lang and then merge the jsons; but the json merge is also fairly complex, it's less explicit, and overall seems to be like a worse solution.

Reply:
Ok, seems legit. Is it worth to have a parser per each lang? I mean, we can rely on one parser and simply pass the list of langs code you want in the `langs` key.

Reply:
Yeah, this point was in fact driven by the implementation...
As per Laurent recommendation, records.with_context(lang=...) can be very costly. 
So putting it this way makes it very easy to only do it the minimum number of times.
To have the more natural way to just put the lang in the parser's dict, the beginning of jsonify would need to extract the translated fields to keep the same efficiency.

Reply:
Ok, but I'm saying this structure is not required IMO.

The question is: how many times you want to have a different result schema by lang? Never, likely :)

Hence, you could change this

```py
parser = {
    "resolver": 3,
    "language_agnostic": True,
    "langs": {
        False: [
            {'name': 'description'},
            {'name': 'number', 'resolver': 5},
            ({'name': 'partner_id', 'alias': 'partners'}, [{'name': 'display_name'}])
        ],
        'fr_FR': [
            {'name': 'description', 'alias': 'descriptions_fr'},
            ({'name': 'partner_id', 'alias': 'partners'}, [{'name': 'description', 'alias': 'description_fr'}])
        ],
    }
}
```
to
```py
parser = {
    "resolver": 3,
    "language_agnostic": False,  # this key can be dropped or made computed on `langs`
    "parser|schema|fields":  [
            {'name': 'description'},
            {'name': 'number', 'resolver': 5},
            ({'name': 'partner_id', 'alias': 'partners'}, [{'name': 'display_name'}])
        ],
    "langs": ["fr_FR", "it_IT"],  # maybe m2m to res.lang in the resolver record
}
```
and then you generate the translated key for each field by appending - by default - the lang code to the key.
You can pass `translate_key_func` to override the behavior (eg: `lang.split("_")[0]`)

The rest of the implementation stays the same, just do not lookup for a resolver for every lang.
Am I missing anything?

Reply:
Oh, I see what you mean.
Well, typically that API we are working with is the annoying kind: any custom fields must be put into extraData* as a {name, value, type} dictionary.
But it's not that rare, I've worked with 3 platforms that allowed custom fields/columns and they all had similar (but different!) requirements.

Reply:
> 3 platforms that allowed custom fields/columns and they all had similar (but different!) requirements.

you mean per lang? Each lang w/ different keys? (eg: FR has "name" and "description" but IT has only "name"?)

Reply:
Sorry, I meant here that for English description the alias should be `Description`, but for other langs the alias should be `extraData*` (and then in a dict with `'name':  description_sale_langcode`). So:

```
{
    "Description": "English description",
    "extraData": [{"name": "description_sale_fr_BE", "value": "Description française", "type": "str"}],
}
```

(and so I meant that some API have a similar concept of extraData, but that would be slightly differently handled)
(also the field to dict transformation is handled by a generic field resolver)

With your proposed solution (which is simpler), this would require to afterwards modify the dict structure to find all translated fields and put them into the extraData* list. It's a bit less explicit and a tad tricky to do (do you go through all _fields that are translatable, or do you pass the list of fields that should go into extraData another way?).